### PR TITLE
🎨 (#123) Cleanup frontend

### LINKED ISSUE #123: Clean up frontend code
- [x] Add types lmao
- [x] Clean up types.ts, lots of unused types here
- [x] Add doc strings

### PR DESCRIPTION
**Pull request by annaiandresen**
## Describe your changes

**TLDR; general cleanup of app**

- `npm run lint` actually works wow
- types were confusing, added some doc string to the interfaces
- moved OWGroup type -> types.ts
- moved types.ts -> /lib
- moved some reused functions to lib/functions
- removed hardcoded styles
- removed unused css selectors
- moved sidebar components into sidebar/
- moved some assets into src/assets
- think i fixed **SOME** type and lint errors
- updated prettier/eslint setup
- added lint + format scripts
- removed mock api files

### NB! 

- ts support is still not present in the html-part of svelte, so we will still receive hints/warnings here with `svelte-check`. eslint seems to allow it tho. 
- `svelte-check` still complains about Flatpickr, but I'm not able to fix this.
- theres a typing in PunishmentRow that I'm not completely sure about. VScode throws a warning here for me.  

## Issue ticket number and link

Resolves #123 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] I have run lints

### CODE REVIEW

**Review by Terbau — APPROVED** [review]
53 files later, lgtm
